Homework 19

TestProGio opened the pull request:
I added a few not required methods to get around obstacles I was encountering.

TestProGio committed:
First Commit on my Fork

Forked a cloned repo from A59 testPro

TestProGio committed:
HW 19

. Create a new branch named homework-19.

2. Create a new file Homework19.java in IntelliJ IDEA.

3. Create a test case, deletePlaylist() using @Test annotation.

4. Use the helper/reusable methods we created.

5. Use @Parameters for passing baseUrl from the TestNG config file to the tests.

6. Navigate to "https://qa.koel.app/".

7. Log in with your credentials.

8. Click the playlist you want to delete.

9. We should see a red button "x PLAYLIST" on the top right part of the page, and click on it.

Note: If the red button is not displayed, let's create a new playlist and click it.

10. Verify that the confirmation notification displayed has the text, "Deleted playlist {playlist name}".

11. Commit your changes to the new branch homework-19.

12. Push your code to the remote repository.

13. Create a pull request.

Berichicko commented on src/test/java/Homework19.java:
```diff
@@ -0,0 +1,88 @@
+
+public class Homework19 extends BaseTest {
+
+    @Test
+    public void deletePlaylist() throws InterruptedException {
+        // Performing actions: login, search song, select 1st result, add to playlist
+        // LOGIN
+        login("giovanna.silva@testpro.io", "ShakaMaya1302!");
+        Thread.sleep(2000);
+
+        // Click to view playlist
+        deletePlaylistAction();
+
+        // Delete the playlist
+        confirmDelete();
+
+        // Confirm Deletion Success
+        confirmDeletionSuccess();
+
+        // Quitting the WebDriver session
+        driver.quit();
```
**Use in test methods, the following logical split   **Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```

Please, add Assert

**One test - One Assert**

TestProGio commented:
Hi,
Thanks for your feedback! In regards to: 
// THEN
Assert.assertTrue(getSuccessPopUp().isDisplayed());

Since I am new to using this "Assert" method, I was very confused on how to use it with a div pop up that appears and disappears within a second confirming the playlist is deleted. I would like to address the request to add the Assert, but I tried and tried and have no idea since the div appears and disappears. Please advice.

Berichicko commented:
> Hi, Thanks for your feedback! In regards to: // THEN Assert.assertTrue(getSuccessPopUp().isDisplayed());
> 
> Since I am new to using this "Assert" method, I was very confused on how to use it with a div pop up that appears and disappears within a second confirming the playlist is deleted. I would like to address the request to add the Assert, but I tried and tried and have no idea since the div appears and disappears. Please advice.

Hi dear

Please, re-submit homework for review

Berichicko commented:
> Hi, Thanks for your feedback! In regards to: // THEN Assert.assertTrue(getSuccessPopUp().isDisplayed());
> 
> Since I am new to using this "Assert" method, I was very confused on how to use it with a div pop up that appears and disappears within a second confirming the playlist is deleted. I would like to address the request to add the Assert, but I tried and tried and have no idea since the div appears and disappears. Please advice.

**I will try to help you how easy it is to find locators:**

For example

 `By.cssSelector("div.success.show")`

![image](https://user-images.githubusercontent.com/104761482/235407463-672c5728-57d6-4eb8-ae1b-945c83f12f5f.png)

![image](https://user-images.githubusercontent.com/104761482/235407647-4457ba8f-eba2-4c8f-8f57-bae764eeba43.png)

Browse to the desired page
1.	Open the dev console - F12 on Windows/Linux or option + ⌘ + J on macOS
2.	Select the Sources tab in chrome inspector
3.	In the web browser window, hover over the desired element to initiate the popover
4.	Hit F8 on Windows/Linux (or fn + F8 on macOS or  CTRL+\) while the popover is showing. If you have clicked anywhere on the actual page F8 will do nothing. Your last click needs to be somewhere in the inspector, like the sources tab
5.	Go to the Elements tab in inspector
6.	Find your popover (it will be nested in the trigger element's HTML)
7.	Have fun modifying the CSS
8.	https://www.howtogeek.com/663950/what-your-function-keys-do-in-chrome-devtools/

TestProGio committed:
Homework-19 v2

I included 'Assert.assertTrue(getSuccessPopUp().isDisplayed());' with your helpful tips. Thanks. Gio
